lib/stack_master/parameter_resolvers/one_password.rb

```diff
@@ -0,0 +1,66 @@
+module StackMaster
+  module ParameterResolvers
+    class OnePassword < Resolver
+      OnePasswordNotFound = Class.new(StandardError)
+
+      array_resolver
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+      end
+
+      def resolve(params={})
+        raise RuntimeError, "1password requires the `OP_SESSION_<name>` to be set" if ENV.keys.grep(/OP_SESSION_\w+$/).empty?    
+        get_items(params)
+      end
+
+      private
+
+      def validate_op_installed?
+        %x(op --version)
+        rescue Errno::ENOENT => exception
+          raise RuntimeError, "The op cli needs to be installed and in the PATH, #{exception}"
+      end
+
+      def parseable_json?(item)
+        JSON.parse(item)
+        rescue TypeError => exception
+          raise "item returned is not valid JSON: #{item}"
+        rescue JSON::ParserError => exception
+          raise "Failed to parse JSON returned, #{item}"
+      end
+
+      def op_get_item(item, vault)
+        validate_op_installed?
+        begin
+          item = %x(op get item --vault='#{vault}' '#{item}' 2>&1)
```

Comment:
What's the reasoning behind redirecting `stderr` to `stdout`?  I doubt `stderr` will be outputting JSON, which is a requirement the line below.  If there was any output on `stderr` this would probably fail.

Better to let it write to `stderr`?  Or send to `/dev/null` if you don't care about it?

Reply:
No, it captures the error message from `op`, which we now handle a bit better